src/Reflection/EntityFieldsViaMagicReflectionExtension.php

```diff
@@ -32,7 +32,7 @@ public function hasProperty(ClassReflection $classReflection, string $propertyNa
             // Content entities have magical __get... so it is kind of true.
             return true;
         }
-        if ($reflection->implementsInterface('Drupal\Core\Field\FieldItemListInterface')) {
+        if ($reflection->getName() === 'Drupal\Core\Field\FieldItemListInterface' || $reflection->implementsInterface('Drupal\Core\Field\FieldItemListInterface')) {
```

Comment:
I think this is a great time to leverage ObjectType here. We can use an object type for the interface and check if the current item is a superOf the interface. If I recall, being the same returns true. Helps streamline the conditions

Reply:
OK, I think I changed it to what you had in mind :)